libs/pq_proto/src/framed.rs

```diff
@@ -214,27 +214,24 @@ where
     }
 }
 
+/// Cancellation safe as long as the AsyncWrite is cancellation safe.
 async fn flush<S: AsyncWrite + Unpin>(
     stream: &mut S,
     write_buf: &mut BytesMut,
 ) -> Result<(), io::Error> {
     while write_buf.has_remaining() {
-        let bytes_written = stream.write(write_buf.chunk()).await?;
+        let bytes_written = stream.write_buf(write_buf).await?;
         if bytes_written == 0 {
             return Err(io::Error::new(
                 ErrorKind::WriteZero,
                 "failed to write message",
             ));
         }
-        // The advanced part will be garbage collected, likely during shifting
-        // data left on next attempt to write to buffer when free space is not
-        // enough.
-        write_buf.advance(bytes_written);
     }
```

Comment:
Was there a reason why `write_buf` was not used originally? @arssher @petuhovskiy

Reply:
Should had not left out the `write_buf.clear()` from the highlight, on the next line.

I think that was extra given how BytesMut reuses the buffer.

Reply:
Toggling auto-merge as I assume there is no reason for not using write_buf as read_buf is used elsewhere. Please do not resolve the conversation too early :)

Reply:
I looked at the BytesMut impl and `write_buf` and `clear` are doing exactly the same, so no reason